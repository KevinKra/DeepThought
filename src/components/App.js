import React from "react";
import LoginPage from "./LoginPage";
class App extends React.Component {
  constructor() {
    super();
    this.state = {
      userName: ""
    };
  }

  updateUsername = newName => {
    this.setState({ userName: newName });
  };
  render() {
    return (
      <LoginPage
        updateUsername={this.updateUsername}
        userName={this.state.userName}
      />
    );
  }
}

export default App;
