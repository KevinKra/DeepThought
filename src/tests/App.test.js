import React from "react";
import ReactDOM from "react-dom";
import App from "../components/App";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const defaultState = {
  userName: ""
};

describe("<App />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it("Should render App", () => {
    console.log(wrapper.debug());
  });
  it("It matches the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("updateUserName method should update the userName local state when invoked", () => {
    expect(wrapper.state()).toEqual(defaultState);
    const mockUsername = "big-smelly-feet";
    wrapper.instance().updateUsername(mockUsername);
    expect(wrapper.state()).toEqual({ userName: mockUsername });
  });
});
