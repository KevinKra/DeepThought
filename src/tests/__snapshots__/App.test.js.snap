// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> It matches the snapshot 1`] = `
<LoginPage
  updateUsername={[Function]}
  userName=""
/>
`;
